Microsoft.Toolkit.Uwp.UI.Controls/Loading/Loading.Properties.cs

```diff
@@ -0,0 +1,116 @@
+            nameof(IsLoading), typeof(bool), typeof(Loading), new PropertyMetadata(default(bool), IsLoadingPropertyChanged));
+
+        /// <summary>
+        /// Identifies the <see cref="LoadingOpacity"/> dependency property.
+        /// </summary>
+        public static readonly DependencyProperty LoadingOpacityProperty = DependencyProperty.Register(
+            nameof(LoadingOpacity), typeof(double), typeof(Loading), new PropertyMetadata(default(double)));
+
+        /// <summary>
+        /// Identifies the <see cref="LoadingBackground"/> dependency property.
+        /// </summary>
+        public static readonly DependencyProperty LoadingBackgroundProperty = DependencyProperty.Register(
+            nameof(LoadingBackground), typeof(SolidColorBrush), typeof(Loading), new PropertyMetadata(default(SolidColorBrush)));
+
+        private Grid _rootGrid;
+        private Grid _backgroundGrid;
+        private ContentPresenter _contentGrid;
+
+        /// <summary>
+        /// Gets or sets how a child element is vertically positioned or stretched.
+        /// </summary>
+        public VerticalAlignment LoadingVerticalAlignment
+        {
+            get { return (VerticalAlignment)GetValue(LoadingVerticalAlignmentProperty); }
+            set { SetValue(LoadingVerticalAlignmentProperty, value); }
+        }
+
+        /// <summary>
+        /// Gets or sets where an element should be displayed on the horizontal axis.
+        /// </summary>
+        public HorizontalAlignment LoadingHorizontalAlignment
+        {
+            get { return (HorizontalAlignment)GetValue(LoadingHorizontalAlignmentProperty); }
+            set { SetValue(LoadingHorizontalAlignmentProperty, value); }
+        }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether isLoading.
+        /// </summary>
+        public bool IsLoading
+        {
+            get { return (bool)GetValue(IsLoadingProperty); }
+            set { SetValue(IsLoadingProperty, value); }
+        }
+
+        /// <summary>
+        /// Gets or sets the opacity for the panel who appears and disappears behind our custom control.
+        /// </summary>
+        public double LoadingOpacity
+        {
+            get { return (double)GetValue(LoadingOpacityProperty); }
+            set { SetValue(LoadingOpacityProperty, value); }
+        }
+
+        /// <summary>
+        /// Gets or sets the background for the panel who appears and disappears behind our custom control.
+        /// </summary>
+        public SolidColorBrush LoadingBackground
+        {
+            get { return (SolidColorBrush)GetValue(LoadingBackgroundProperty); }
+            set { SetValue(LoadingBackgroundProperty, value); }
+        }
+
+        private static void IsLoadingPropertyChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            var control = (d as Loading);
+            control?.OnApplyTemplate();
```

Comment:
Do not call OnApplyTemplate. Call a method that will adjust based on IsLoading changing. This method should also be called from OnApplyTemplate